File: problems/806-number-of-lines-to-write-string/solution.py
class Solution:
    def numberOfLines(self, widths: List[int], s: str) -> List[int]:
        lines = 1
        width = 0
        for char in s:
            char_width = widths[ord(char) - ord('a')]
            if width + char_width <= 100:
                width += char_width
            else:
                lines += 1
                width = char_width
        return [lines, width]

File: problems/806-number-of-lines-to-write-string/solution.cpp
class Solution {
public:
    vector<int> numberOfLines(vector<int>& widths, string s) {
        int lines = 1;
        int width = 0;
        for (char c : s) {
            int w = widths[c - 'a'];
            if (width + w <= 100) {
                width += w;
            } else {
                lines++;
                width = w;
            }
        }
        return {lines, width};
    }
};